TemperatureMeter.c

```diff
@@ -0,0 +1,102 @@
+/*
+htop - TemperatureMeter.c
+(C) 2013 Ralf Stemmer
+(C) 2014 Blair Bonnett
+Released under the GNU GPL, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "TemperatureMeter.h"
+
+#include "ProcessList.h"
+#include "CRT.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+
+/*{
+#include "Meter.h"
+}*/
+
+int TemperatureMeter_attributes[] = {
+   TEMPERATURE_COOL,
+   TEMPERATURE_MEDIUM,
+   TEMPERATURE_HOT,
+};
+
+static void TemperatureMeter_setValues(Meter* this, char* buffer, int len) {
+   ProcessList* pl = this->pl;
+   this->total = pl->totalTasks;
+   this->values[0] = pl->runningTasks;
+   snprintf(buffer, len, "%d/%d", (int) this->values[0], (int) this->total);
+}
+
+static void TemperatureMeter_display(Object* cast, RichString* out) {
+   (void) cast;
+   FILE *p;
+   p = popen("sensors", "r");
```

Comment:
Is there a non-exec way of getting the cup temperature portable?
(`/sys/class/thermal/thermal_zone0/temp`)

Reply:
Maybe. I used this path to get temperature of my Raspberry Pi (htop-temperature-rpi in AUR). But I believe it'll be limited or may somehow not work on some systems, like `sensors` not working on some of them.

Reply:
For example, I have on my ASUS UX433FN ten thermal zones. But on other notebooks and devices they can differ. And `sensors` covers almost all of them.